### DISCUSSION TITLE
Go language bindings

### DISCUSSION THREAD

**Discussion by julianthome** [tree-sitter/tree-sitter#1020] [Ideas] [answered]
At the moment, tree-sitter does not have official language bindings for Go. I am curious whether the tree-sitter community would be interested in having official Go support.

I am aware of https://github.com/smacker/go-tree-sitter but I believe this can be rather considered to be a generic parsing project than a language-binding implementation given that they included the tree-sitter sources and the grammar sources.

Are there any conditions for new language bindings to be officially listed on https://tree-sitter.github.io/tree-sitter/#language-bindings and/or integrated as a sub-project?

**Comment by ahlinc** [comment]
@julianthome I think there are no strict conditions, but it would be great if such binding will be well tested and have a good coverage of tree-sitter's C API facilities. For example `OCaml` binding listed there but maintained outside of Tree-sitter's org.

**Comment by maxbrunsfeld** [comment]
AFAICT, the actual bindings in https://github.com/smacker/go-tree-sitter are very high-quality, but I agree that the project isn't as modular as the other binding libraries, because it bundles the grammars.

The challenge in modularizing it will be figuring out how to make individual grammars consumable from Go. It might require including some additional (generated) Go-specific source files in each grammar repo, similar to what we do for Rust and Node.js. I'm open to adding a `bindings/go` directory that would be populated automatically by `tree-sitter generate`.

**Comment by ObserverOfTime** [comment, marked as answer]
https://github.com/tree-sitter/go-tree-sitter
